sys/dev/drm/bridges/tda19988/tda19988.c

```diff
@@ -590,11 +590,6 @@ tda19988_start(struct tda19988_softc *sc)
 
 	tda19988_cec_write(sc, TDA_CEC_FRO_IM_CLK_CTRL,
 	    CEC_FRO_IM_CLK_CTRL_GHOST_DIS | CEC_FRO_IM_CLK_CTRL_IMCLK_SEL);
-
-	/* Default values for RGB 4:4:4 mapping */
-	tda19988_reg_write(sc, TDA_VIP_CNTRL_0, 0x23);
-	tda19988_reg_write(sc, TDA_VIP_CNTRL_1, 0x01);
-	tda19988_reg_write(sc, TDA_VIP_CNTRL_2, 0x45);
```

Comment:
These should come from the video-ports property of the FDT node:
```
		ret = of_property_read_u32(np, "video-ports", &video);
		if (ret == 0) {
			priv->vip_cntrl_0 = video >> 16;
			priv->vip_cntrl_1 = video >> 8;
			priv->vip_cntrl_2 = video;
		}
```
from Linux, where video-ports is `0x234501`. Which means I think you just had CNTRL_1 and _2 round the wrong way for these hard-coded values for Morello? Note that the YAML binding specifies a default of 0x230145, which is what you put here.

Reply:
And indeed 0x234501 is how the Morello EDK2 configures the driver: https://git.morello-project.org/morello/edk2-platforms/-/blob/morello/master/Platform/ARM/Morello/Drivers/Tda19988Dxe/Driver.c#L1203-1216 (hence why it "works" when you don't set the values, I guess, relying on firmware to have done it)

Reply:
More complete fix in #1894